projects/packages/sync/src/class-listener.php

```diff
@@ -480,4 +484,56 @@ public function get_request_url() {
 		}
 		return is_admin() ? get_admin_url( get_current_blog_id() ) : home_url();
 	}
+
+	/**
+	 * Check the size of the data for a given Sync action, to avoid adding them
+	 * to the Sync queue if they exceed a certain limit.
+	 *
+	 * @param  array  $args        The data associated with a given Sync action.
+	 * @param  string $action_name The Sync action name.
+	 * @return bool                True if the data size is under the allowed limit, false otherwise.
+	 */
+	protected function check_data_size_by_action( $args, $action_name ) {
+		if ( false === array_key_exists( $action_name, Defaults::$default_max_args_size_per_action ) ) {
+			return true;
+		}
+
+		$max_args_size = null;
+
+		$max_args_size_per_action = Defaults::$default_max_args_size_per_action[ $action_name ];
+
+		if ( ! empty( $max_args_size_per_action['exceptions'] ) ) {
+			$exceptions = $max_args_size_per_action['exceptions'];
+			$name       = $args[0];
+			foreach ( $exceptions as $name_or_prefix => $max_size ) {
+				if ( $name === $name_or_prefix ) {
+					$max_args_size = $max_size;
+					break;
+				}
+				$is_prefix = '*' === substr( $name_or_prefix, -1 );
+				if ( $is_prefix ) {
+					$name_starts_with = substr( $name_or_prefix, 0, -1 );
+					if ( strpos( $name, $name_starts_with ) === 0 ) {
+						$max_args_size = $max_size;
+						break;
+					}
+				}
+			}
+		}
+
+		$max_args_size = isset( $max_args_size ) ? $max_args_size : $max_args_size_per_action['default'];
+		try {
+			$serialized_args = serialize( $args ); // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
+		} catch ( Exception $e ) {
+			// Possible case would be a callable returning an anonymous function.
+			return true;
```

Comment:
@bisko Should we maybe return `false` here? Do we allow that?

Reply:
Just an FYI, after discussing this we decided to return `false` instead since we don't allow callables returning an anonymous function